### ISSUE TITLE
Seeking help of "npm run build"

### ISSUE THREAD

**Issue by Gracidora** [gfxfundamentals/threejsfundamentals#140]
I'm learning three.js and I happened to find these articles one day. 
These articles are so great that I want to contribute some translations !

1. But I met trouble when I proceeded to "npm run build", the warning is like this:

   `ERROR
Warning: Unable to delete "out/3rdparty" file (EACCES: permission denied, rmdir 'out/3rdparty'). Use --force to continue.`

2. When I tried "npm run build --force", the warning was the same.

3. When I tried "sudo npm run build", the warning became :
`Warning: Cannot find module '../build/Release/canvas.node'`

What did I do wrong? How to fix it？
Thanks

**Comment by greggman** [comment]
Sorry you ran into issues and thank you for the offer to help translate

I don't know what went wrong. It sounds like maybe some commands were run with `sudo`?

You're on linux or macos it looks like. First you should probably install node with [nvm](https://github.com/nvm-sh/nvm). 

Then start new

```
git clone https://github.com/gfxfundamentals/threejsfundamentals.git
cd threejsfundamentals
npm install
npm run build
```

You should never use `sudo` to run `npm run build` and if you use `nvm` you should never need to run any node stuff with `sudo`

Otherwise you can try to delete the out folder. Since you're getting permission errors I can only guess it was created by some sudo run so you'll have to use sudo to delete it

```
sudo rm -rf out
```

But I suspect a bunch of other files are probably also set wrong at this point which is why I suggest starting over.

Also, what version of node? Use `node --version`

I've built with node 12.18.0 and with node 14.13.1. I just tried it now, cloning the repo and following the steps above to make sure it works. Hopefully it will work for you too.

**Comment by Gracidora** [comment]
Thanks a lot for your help. I deleted all the related files and switched node version to 14.13.1, and met other trouble. 


- I tried "npm install" , the warning went as 

`node-pre-gyp WARN Using needle for node-pre-gyp https download
`
I searched for this in website ,someone suggested using `--build-from-source`.


- I tried `npm install --build-from-source`  then, the warning went like 

```
node-pre-gyp WARN Using needle for node-pre-gyp https download 
gyp WARN EACCES current user ("mac") does not have permission to access the dev dir "/Users/mac/Library/Caches/node-gyp/14.13.1"

```

Should I use `sudo` or  `--build-from-source` in  `npm install` ?
